### PR TITLE
Chengyan phase2 update reusable type backend

### PR DESCRIPTION
**Pull request by ChengyanOo**
# Description
This update introduces a backend endpoint addition for the new reusable type. It ensures seamless integration with the frontend's item type management functionalities.

## Related PRs (if any):
While the frontend integration for this feature is still under development, you can utilize the in-progress frontend work for testing purposes. Check out [Chengyan phase2 reusables list UI #1956](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1956/) for the frontend changes. This will allow you to test the backend functionality in a more interactive manner, keeping in mind that the frontend is not yet finalized and subject to change.

## Main changes explained:
- Implemented the two new REST endpoints: `http://localhost:4500/api/bm/invtypes/reusables`, and `http://localhost:4500/api/bm/invtypes/reusable`

## How to test:
1. Switch to the current branch.
2. Execute `npm install`, `npm run build`, and `npm run start` to run this PR locally.
3. Test in conjunction with the frontend PR [Chengyan phase2 reusables list UI #1956](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1956/)
4. Watch the detailed walkthrough video for recommended way to test.
5. Review the code for potential code smells.

## Screenshots or videos of changes:
Screenshot of Feature
Detailed walkthrough: [https://www.loom.com/share/cedd7377335845738e0eac02169fb696?sid=286fb8d4-24a2-4d21-a1dd-8584d4a3614b](https://www.loom.com/share/cedd7377335845738e0eac02169fb696?sid=286fb8d4-24a2-4d21-a1dd-8584d4a3614b)

### CODE REVIEW

File: src/controllers/bmdashboard/bmReusableController.js [hunk 1]
```diff
@@ -47,17 +47,10 @@ const bmReusableController = function (BuildingReusable) {
         }
     };
 
-    const purchaseReusable = async (req, res) => {
```
Comment (ImzIssa) [inline src/controllers/bmdashboard/bmReusableController.js:50]: Hey, I am not able to build the app. looks like you deleted purchaseReusable but its still being return from the controller which is causing an error.

PR 773

**Review by ImzIssa — CHANGES_REQUESTED** [review]
Hey I am getting this error when I try to build.
  
PR 773

**Review by Sandhya1236 — CHANGES_REQUESTED** [review]
The application crashes with a ReferenceError: purchaseReusable is not defined in bmReusableController.js. This occurs because the purchaseReusable variable or function is referenced without being declared or imported. Please fix this.

image

**Review by Infinite-Turkey — CHANGES_REQUESTED** [review]
image
The purchaseReusable variable or function is being used in the bmReusableController.js file, but it hasn't been declared or imported.

**Review by Wenboooooo — CHANGES_REQUESTED** [review]
Hello, it seems like the backend pulled from the branch cannot be run.
image

**Review by Kavil-Jain-514 — CHANGES_REQUESTED** [review]
Hey there, I am unable to start the back-end. Error:
image

**Pull request closed without merging**